service.js

```diff
@@ -0,0 +1,86 @@
+import request from 'request';
+import uuid from 'node-uuid';
+import promisify from './promisify';
+import ConsultantError from './consultant-error';
+
+const deregister = async (host, instance) => {
+	const req = {
+		url : `${host}/v1/agent/service/deregister/${instance.instance}`,
+		method : 'DELETE',
+		headers : {
+			['user-agent'] : 'consultant-js'
+		}
+	};
+
+	try {
+		await promisify(cb => request(req, cb));
+	}
+	catch (e) {
+		throw new ConsultantError('Could not deregister from Consul');
+	}
+};
+
+export default async function register({service:{port, name, dataCenter, host, instance}, healthCheckPath, healthCheckInterval = 10, consulHost}) {
+	consulHost = consulHost || process.env.CONSUL_HOST || 'http://localhost:8500';
+
+	if (!port) {
+		throw new ConsultantError('port was not defined');
+	}
+
+	if (!name) {
+		throw new ConsultantError('service.name was not defined');
+	}
+
+	let agent = null;
+	try {
+		const res = await promisify(cb => request({uri : `${consulHost}/v1/agent/self`, timeout : 5000}, cb));
+		agent = JSON.parse(res.body);
+	}
+	catch (e) {
+		throw new ConsultantError('Could not retrieve agent from consul');
+	}
+
+	const identifier = {
+		name : name || process.env.SERVICE_NAME,
+		dataCenter : dataCenter || process.env.SERVICE_DC || agent.Config.Datacenter,
+		host : host || process.env.SERVICE_HOST || agent.Config.NodeName,
+		instance : instance || process.env.SERVICE_INSTANCE || uuid.v4()
+	};
+
+	const body = {
+		['Id'] : identifier.instance,
```

Comment:
The keys are doubly unnecessary verbose. This can just be `const body = { Id : identifier.instance, ...}`. Same goes at more places in this PR. You only need to quote when there are special characters in the string (such as in `'user-agent'`), but even then you don't need the `[]`. Those are only needed for computed property names.